src/Dependencies/Collections/SegmentedArray.cs

```diff
@@ -394,6 +394,22 @@ public static void Sort<T>(SegmentedArray<T> array, int index, int length, IComp
             }
         }
 
+        public static void Sort<T>(SegmentedArray<T> array, int index, int length, Comparison<T> comparison)
```

Comment:
copy of method above, just taking and passing through a Comparison instead of an IComparer.  See https://github.com/dotnet/roslyn/pull/73648/files#r1610644205 for more details.